src/game/AI/ScriptDevAI/scripts/northrend/naxxramas/boss_anubrekhan.cpp

```diff
@@ -21,127 +21,105 @@ SDComment: Intro text usage is not very clear. Requires additional research.
-    SAY_SLAY                    = -1533008,
-
-    EMOTE_CRYPT_GUARD           = -1533153,
-    EMOTE_INSECT_SWARM          = -1533154,
-    EMOTE_CORPSE_SCARABS        = -1533155,
+    SAY_GREET                   = 13004,
+    SAY_AGGRO1                  = 13000,
+    SAY_AGGRO2                  = 13002,
+    SAY_AGGRO3                  = 13003,
+    SAY_TAUNT1                  = 13006,
+    SAY_TAUNT2                  = 13007,
+    SAY_TAUNT3                  = 13008,
+    SAY_TAUNT4                  = 13009,
+    SAY_SLAY                    = 13005,
+
+    EMOTE_CRYPT_GUARD           = 29887,
+    EMOTE_INSECT_SWARM          = 13443,
+    EMOTE_CORPSE_SCARABS        = 32796,
 
     SPELL_DOUBLE_ATTACK         = 18943,
     SPELL_IMPALE                = 28783,                    // May be wrong spell id. Causes more dmg than I expect
     SPELL_IMPALE_H              = 56090,
     SPELL_LOCUSTSWARM           = 28785,                    // This is a self buff that triggers the dmg debuff
     SPELL_LOCUSTSWARM_H         = 54021,
 
-    // spellId invalid
-    // SPELL_SUMMONGUARD         = 29508,                   // Summons 1 crypt guard at targeted location - spell doesn't exist in 3.x.x
+    SPELL_SUMMON_GUARD          = 29508,                   // Summons 1 crypt guard at targeted location - spell doesn't exist in 3.x.x
 
     SPELL_SELF_SPAWN_5          = 29105,                    // This spawns 5 corpse scarabs ontop of us (most likely the pPlayer casts this on death)
     SPELL_SELF_SPAWN_10         = 28864,                    // This is used by the crypt guards when they die
 
-    NPC_CRYPT_GUARD             = 16573
+    NPC_CRYPT_GUARD             = 16573,
+
+    SPELLSET_10N                = 1595601,
+    SPELLSET_25N                = 2924901,
 };
 
-static const DialogueEntry aIntroDialogue[] =
+enum AnubRekhanActions
 {
-    {SAY_GREET,   NPC_ANUB_REKHAN,  7000},
-    {SAY_TAUNT1,  NPC_ANUB_REKHAN,  13000},
-    {SAY_TAUNT2,  NPC_ANUB_REKHAN,  11000},
-    {SAY_TAUNT3,  NPC_ANUB_REKHAN,  10000},
-    {SAY_TAUNT4,  NPC_ANUB_REKHAN,  0},
-    {0, 0, 0}
+    ANUBREKHAN_ACTIONS_MAX,
+    ANUBREKHAN_SUMMON,
 };
 
-static const float aCryptGuardLoc[4] = {3333.5f, -3475.9f, 287.1f, 3.17f};
-
-struct boss_anubrekhanAI : public ScriptedAI
+struct boss_anubrekhanAI : public BossAI
 {
-    boss_anubrekhanAI(Creature* pCreature) : ScriptedAI(pCreature),
-        m_introDialogue(aIntroDialogue)
+    boss_anubrekhanAI(Creature* creature) : BossAI(creature, ANUBREKHAN_ACTIONS_MAX),
+        m_instance(static_cast<instance_naxxramas*>(creature->GetInstanceData())),
+        m_isRegularMode(creature->GetMap()->IsRegularDifficulty())
     {
-        m_pInstance = (instance_naxxramas*)pCreature->GetInstanceData();
-        m_bIsRegularMode = pCreature->GetMap()->IsRegularDifficulty();
-        m_introDialogue.InitializeDialogueHelper(m_pInstance);
-        m_bHasTaunted = false;
+        m_hasTaunted = false;
```

Comment:
ykno, u can also set m_hasTaunted the same way

Reply:
I know, but I like setting variables that are used in multiple scripts in the format above, and unique variables in the format below